egi2bids/mff2bids.py

```diff
@@ -47,80 +57,142 @@
 # fmt: on
 
 
+def _extract_folder(
+    file: Union[str, Path], dir_: Union[str, Path] = None
+) -> Path:
+    """Extract a .mff compressed folder to its original form."""
+    # check paths and file extension
+    file = _ensure_path(dir_, must_exist=True)
+    ext = file.suffix
+    _check_value(ext, (".tar", ".zip", ".mff"), "extension")
+    dir_ = Path.cwd() if dir_ is None else dir_
+    dir_ = _ensure_path(dir_, must_exist=True)
+    # open the archive if needed
+    archive_readers = {
+        ".tar": tarfile.open,
+        ".zip": zipfile.ZipFile,
+    }
+    if ext in (".tar", ".zip"):
+        logger.info("Extracting '%s' archive %s to %s.", ext, file, dir_)
+        with archive_readers[ext](file, "r") as archive:
+            archive.extractall(dir_)
+        for root, dirs, _ in os.walk(dir):
+            if "Contents" in dirs:
+                logger.info("MFF file found in %s", root)
+                return Path(root)
+        else:
+            raise (f"The '{ext}' archive does not contain a 'Content' folder.")
+
+    elif ext == ".mff":
+        return file
+
+
+@verbose
 def mff2bids(
-    mff_source,
-    bids_root,
-    subject=None,
-    task=None,
-    session=None,
+    mff_source: Union[str, Path],
+    bids_root: Union[str, Path],
+    subject,
+    session,
+    task,
     run=None,
     event_id=None,
-    line_frequency=50,
+    save_source: bool = False,
+    working_dir=None,
+    *,
+    overwrite: bool = False,
+    verbose=None,
 ):
-    bids_path = BIDSPath(root=Path(bids_root))
-    bids_path.update(subject=subject)
-    bids_path.update(session=session)
-    bids_path.update(task=task)
-    bids_path.update(datatype="eeg")
-    bids_path.update(run=run)
-
-    print(Path(mff_source))
-
-    # load EEG data
-    raw = mne.io.read_raw_egi(Path(mff_source), preload=True)
-    raw.info["line_freq"] = line_frequency
+    logger.info("Processing %s", mff_source)
+    working_dir = (
+        tempfile.TemporaryDirectory(suffix=".mff")
+        if working_dir is None
+        else working_dir
+    )
+    with working_dir as wd:
+        mff_source = _extract_folder(mff_source, dir_=wd)
 
-    # rename channels
-    new_chs = {}
-    for i, ch in enumerate(raw.info["ch_names"]):
-        if i > 256:
-            break
-        new_chs[ch] = ch_names_egi[i]
-    raw.rename_channels(new_chs)
+        # BIDS root
+        bids_root = _ensure_path(bids_root, must_exist=True)
```

Comment:
` must_exist=False` we should allow for the creation of a new BIDS dataset.

Reply:
Agree.